src/syntax_util.ml

```diff
@@ -49,6 +49,37 @@ let replace_string old_str new_str str =
   replace_string_ old_str new_str 0 str buffer;
   Buffer.contents buffer
 
+let split str ~by =
+  let str = ref str in
+  let accum = ref [] in
+  try
+    while true do
+      let strVal = !str in
+      let foundIdx = String.index_from strVal 0 by in
+      let subStr = String.sub strVal 0 foundIdx in
+      str := String.sub strVal (foundIdx + 1) (String.length strVal - foundIdx - 1);
```

Comment:
There is not need to allocate this intermediate string. It makes the function quadratic.